go-controller/pkg/config/config.go

```diff
@@ -1149,6 +1173,21 @@ func pathExists(path string) bool {
 	return true
 }
 
+// joinSubnetOverlap returns an err if joinSubnet range contains the subnet
+func overlapsWithJoinSubnet(subnets []*net.IPNet) error {
+	joinSubnet := V4JoinSubnet
+	if IPv6Mode {
+		joinSubnet = V6JoinSubnet
+	}
+	_, joinIPNet, _ := net.ParseCIDR(joinSubnet)
+	for _, subnet := range subnets {
+		if joinIPNet.Contains(subnet.IP) {
```

Comment:
oh, and actually, this test isn't sufficient; you have to check if `subnet.Contains(joinIP)` too. (eg, if someone tries to use `100.0.0.0/8` as a subnet, that overlaps `100.64.0.0/16`). There's already a function that does this for you: `cidrsOverlap` in `pkg/config/utils.go`.

Reply:
Good Catch. I didn't notice while reviewing. Will fix it.